### PR TITLE
.gitignore: Add generated files

### PR DESCRIPTION
**Pull request by aknuds1**
###  What have you changed?
Add files generated by tikv-server, such as last_tikv.toml to .gitignore, since they shouldn't be tracked by Git.

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

Manually.

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No.

###  Does this PR affect `tidb-ansible`?
No.

### CODE REVIEW

**Comment by siddontang** [comment]
em, I guess the stale files and folders are caused by starting TiKV without assigning a data dir, right? 

seem we need to fix the problem when wrongly running `tikv-server`. /cc @zhangjinpeng1987

File: .gitignore [hunk 1]
```diff
@@ -31,3 +31,7 @@ fuzz-incremental/
 # cargo configuration. We presently use this to create custom cargo profiles
 # that should not be checked in at this location.
 .cargo/
+LOCK
+db/
```
Comment (siddontang) [inline .gitignore]: if we add a same name folder like src/raft, can it be ignored too?

**Comment by aknuds1** [comment]
> em, I guess the stale files and folders are caused by starting TiKV without assigning a data dir, right?
> 
> seem we need to fix the problem when wrongly running `tikv-server`. /cc @zhangjinpeng1987

So it shouldn't be possible to run tikv-server without assigning a data dir? Would you like me to issue a PR to require a data dir to be configured?

File: .gitignore (hunk 1 above)
Comment (aknuds1) [reply .gitignore]: @siddontang If it's better, I can qualify the patterns to be at the root, like `/raft/` instead.

**Comment by siddontang** [comment]
Thanks @aknuds1 

another problem is that if we start TiKV unsuccessfully (may first start TiKV), it may leave the stale files and folders. 

So this is not only for not assigning data dir, if we pass an invalid config or not pass the config file, it may also lead to this problem. 

maybe we must check the config invalid at first then go on.

File: .gitignore (hunk 1 above)
Comment (aknuds1) [reply .gitignore]: @siddontang I changed the PR to qualify the ignore patterns as being in the root, f.ex. /LOCK.

**Pull request merged**